ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/transaction/UtxoFilteredTransactionBuilder.kt

```diff
@@ -0,0 +1,68 @@
+package net.corda.v5.ledger.utxo.transaction
+
+import net.corda.v5.base.annotations.Suspendable
+import net.corda.v5.ledger.utxo.Command
+import net.corda.v5.ledger.utxo.ContractState
+import net.corda.v5.ledger.utxo.StateRef
+import java.security.PublicKey
+import java.util.function.Predicate
+
+interface UtxoFilteredTransactionBuilder {
+
+    // and time window
+    // no benefit of predicates for inputs
+    @Suspendable
+    fun withNotary(): UtxoFilteredTransactionBuilder
+
+    @Suspendable
+    fun withSignatoriesSize(): UtxoFilteredTransactionBuilder
+
+    @Suspendable
+    fun withSignatories(): UtxoFilteredTransactionBuilder
+
+    @Suspendable
+    fun withSignatories(predicate: Predicate<PublicKey>): UtxoFilteredTransactionBuilder
+
+    @Suspendable
+    fun withInputStatesSize(): UtxoFilteredTransactionBuilder
+
+    @Suspendable
+    fun withInputStates(): UtxoFilteredTransactionBuilder
+
+    @Suspendable
+    fun withInputStates(predicate: Predicate<StateRef>): UtxoFilteredTransactionBuilder
+
+    @Suspendable
+    fun withReferenceInputStatesSize(): UtxoFilteredTransactionBuilder
+
+    @Suspendable
+    fun withReferenceInputStates(): UtxoFilteredTransactionBuilder
+
+    @Suspendable
+    fun withReferenceInputStates(predicate: Predicate<StateRef>): UtxoFilteredTransactionBuilder
+
+    @Suspendable
+    fun withOutputStatesSize(): UtxoFilteredTransactionBuilder
+
+    // add with output states infos (for type info)
+    // flag for with all output info or only kept info
+    // TODO Think about emc
```

Comment:
What is this todo?

Reply:
Was for "Encumbrance" but when I was on a call with matthew I realised I didn't know how to spell it 🤣

Reply:
Hmm - the UtxoFilteredTransaction does expose the output/output info split - you either can see the output, or you can't, not way to just see the info. So not sure that makes sense for the filtering.